run.py

```diff
@@ -1,116 +1,441 @@
+            print 'trying deeper probe'
+
+        # Now let's call the common check list
+        self._troubleshoot_mon_common()
+
+    def _restart_all_machines(self):
+        for machine in self.machines:
+            if machine.ssh_status == 'LIVE':
+                self._restart_ceph_mon_service('start', machine.host)
+
+    def _troubleshoot_mon_cli(self):
+        '''
+            Troubleshoot mon issues using ceph cli i.e. the quorum has been
+            established
+        '''
+        print 'MON Status : ', cluster_status
+        print '\nProbable cause Ceph mon service not running in some machines'
+        print 'Try & start service in the machines not in quorum',
+        print '(yes/no) (default no)?',
+        response = raw_input()
 
-        if cluster_status in self.BAD_HEALTH:
-            print 'MON Status : ', cluster_status
-            print '\nProbable cause Ceph service not running in some machines'
-            print 'Try & start service in the machines not in quorum',
+        if response != 'yes':
+            print 'not proceeding with updating machines, aborting'
+            return
+
+        # Start ceph cli check_list
+        self._restart_dead_mon_daemons()
+
+        # Let's wait for sometime before getting ceph health status
+        time.sleep(10)
+        try:
+            self.check_ceph_cli_health(self.connection)
+        except TimeoutError:
+            print "Restarting servers not in quorum didn't work,"
+            print 'trying deeper probe'
+
+        # Now let's call the common checklist
+        self._troubleshoot_mon_common()
+
+    def _troubleshoot_mon_common(self):
+        ''' Common checklist for both when ceph cli is working or not '''
+        print 'Trying to detect for Clock Skew'
+        status = None
+        try:
+            status = self._detect_clock_skew(self.connection)
+        except TimeoutError:
+            print 'ceph health command did not work proceeding to next phase'
+
+        if status is None:
+            print 'Could not detect any Clock Skew, proceeding to deeper probe'
+        else:
+            print 'Clock Skew detected for', status, 'Try start ntp server?'
+            print 'We assume ntpd is installed here'
             print '(yes/no) (default no)?',
             response = raw_input()
 
-            if response in ['no', '']:
-                print 'not proceeding with updating machines, aborting'
-                return
-            if response not in ['yes']:
-                print 'response not valid, aborting'
-                return
+            if response != 'yes':
+                print 'aborting'
+                exit()
+            else:
+                self._correct_skew(status)
 
-            command = 'sudo ceph mon_status'
-            (output, err) = self._execute_command(self.connection, command)
+        # Let's wait for sometime before getting ceph health status
+        time.sleep(10)
```

Comment:
It's prob better to poll here for the status you want instead of waiting a certain amount of time